cloudify_deployment_proxy/__init__.py

```diff
@@ -448,12 +450,15 @@ def execute_workflow(self):
 
             ctx.logger.debug('Polling execution succeeded')
 
-        if NIP_TYPE == ctx.node.type:
-            ctx.logger.debug('Start post execute node proxy')
+        type_hierarchy = ctx.node.type_hierarchy
+        if NIP_TYPE in type_hierarchy:
+            ctx.logger.info('Start post execute node proxy')
             return self.post_execute_node_instance_proxy()
-        elif DEP_TYPE == ctx.node.type:
-            ctx.logger.debug('Start post execute deployment proxy')
+
+        elif DEP_TYPE in type_hierarchy:
+            ctx.logger.info('Start post execute deployment proxy')
             return self.post_execute_deployment_proxy()
+
         return False
```

Comment:
I think we should raise here and say something is wrong.

Reply:
actually I think you are right, returning false does not make any sense here since the flow will continue to work as expected without any problem, I will update this and make required changes to unit tests as well